SU2_CFD/src/fluid/CFluidScalar.cpp

```diff
@@ -0,0 +1,129 @@
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cmath>
+#include <math.h>
+#include <vector>
+#include <numeric>
+#include <iostream>
+
+#include "../../include/fluid/CFluidScalar.hpp"
+#include "../../include/fluid/CConstantViscosity.hpp"
+#include "../../include/fluid/CSutherland.hpp"
+#include "../../include/fluid/CPolynomialViscosity.hpp"
+#include "../../include/fluid/CConstantConductivity.hpp"
+#include "../../include/fluid/CConstantConductivityRANS.hpp"
+#include "../../include/fluid/CConstantPrandtl.hpp"
+#include "../../include/fluid/CConstantPrandtlRANS.hpp"
+#include "../../include/fluid/CPolynomialConductivity.hpp"
+#include "../../include/fluid/CPolynomialConductivityRANS.hpp"
+#include "../../include/fluid/CIncIdealGas.hpp"
+
+
+CFluidScalar::CFluidScalar(su2double val_Cp, su2double val_gas_constant, const su2double value_pressure_operating, CConfig *config) : CFluidModel() {
+  
+  n_scalars = config->GetnSpecies();
+  n_species_mixture = n_scalars + 1;
+
+  specificHeat.resize(n_species_mixture);
+  molarMasses.resize(n_species_mixture);
+  massFractions.resize(n_species_mixture);
+  moleFractions.resize(n_species_mixture);
+  laminarViscosity.resize(n_species_mixture);
+  laminarthermalConductivity.resize(n_species_mixture);
+
+  for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+    molarMasses.at(iVar) = config->GetMolecular_Weight(iVar);
+    specificHeat.at(iVar) = 1004.703; //config->GetSpecific_Heat_Cp(iVar);
+  }
+
+  wilke = false;
+  davidson = true;
+
+  Pressure_Thermodynamic = value_pressure_operating;
+  Gas_Constant = val_gas_constant; //config->GetGas_Constant();
+  Gamma = 1.0;
+  Cp = val_Cp;
+  Cv = Cp;
+  SetLaminarViscosityModel(config);
+  SetThermalConductivityModel(config);
+}
+
+void CFluidScalar::SetLaminarViscosityModel(const CConfig* config) {
+  switch (config->GetKind_ViscosityModel()) {
+    case VISCOSITYMODEL::CONSTANT:
+      LaminarViscosity = unique_ptr<CConstantViscosity>(new CConstantViscosity(config->GetMu_ConstantND()));
+      break;
+    case VISCOSITYMODEL::SUTHERLAND:
+      LaminarViscosity = unique_ptr<CSutherland>(
+          new CSutherland(config->GetMu_RefND(), config->GetMu_Temperature_RefND(), config->GetMu_SND()));
```

Comment:
This and `SetThermalConductivityModel` also looks the same as the parent class versions

Reply:
Yes, I have removed them from the CFluidScalar.cpp, Thanks!!

Reply:
great, we will do the PR for the mixing viscosity after this PR.